### DISCUSSION TITLE
QUIC support?

### DISCUSSION THREAD

**Discussion by reececomo** [geckosio/geckos.io#262] [General]
Looks like [socket.io just added QUIC support](https://github.com/socketio/socket.io/issues/3769#issuecomment-1603856833)

Any chance of that being supported by geckos.io?

**Comment by yandeu** [comment]
Thanks.
Node.js still doesn't have native QUIC support. Looks like they are using the `@fails-components/webtransport` package. I will have a look.

**Comment by reececomo** [comment]
Would be curious to see what the performance / packet overhead difference is.

[@darrachquesne](https://github.com/darrachequesne) just published https://socket.io/get-started/webtransport → which details an connection "upgrade" flow (polling → ws → webtransport). Interesting...

**Comment by trekze** [comment]
The socketio support seems a bit convoluted, and not working for me. So this could be a good opportunity for geckos.io to gain an advantage over socketio, if you manage to build WebTransport support with better DX.

**Reply by reececomo** [reply]
The core benefit right now that socket.io has is that it can fallback to other protocols/transport based on device support.

**Reply by reececomo** [reply]
Also no WebRTC ICE

**Comment by reececomo** [comment]
Update: I had a little play around with webtransport and it wasn't a noticeable improvement (also no Safari support right now).

I think geckos.io/wrtc is still the happy path
